### PR TITLE
[T5A1][W15-B4]Jasper Lee

### PR DESCRIPTION
**Pull request by jasperlee27**

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -74,12 +75,36 @@ Use case ends.
...
 
+#### Use case: Edit Tag
+  
+ **MSS** 
+
+1. User requests to edit tag of person.
+2. AddressBook displays person and existing tag.
+3. User requests to change tag e.g 'friends' to 'buddies'
+4. AddressBook asks for confirmation of change tag.
+5. User confirms with 'Y'.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: it is good that you are specific about what is the confirmation command; however, at the stage of use case, you don't need to be so specific  :-)

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -74,12 +75,36 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: Edit Tag
+  
+ **MSS** 
+
+1. User requests to edit tag of person.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: the activity was updated to rename a tag such that all people with the given tag will have the tag updated. please check the description again

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -74,12 +75,36 @@ Use case ends.
...
+> AddressBook shows an error message <br>
+ 	Use case resumes at step 2.
+ 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should have input validation. Prevent malicious input to crash AddressBook (e.g hasNext() delimiter)
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: like we discussed in the tutorial, this is a gray area, does it belong to functional requirements or NFR. but it is good to be captured here than not captured at all :-)

**Comment by okkhoy** [comment]
@jasperlee27 
Some comments added. Please ack & close the PR after reading comments.

**Comment by jasperlee27** [comment]
Acknowledged

**Pull request closed without merging**